plugins/CoreVisualizations/javascripts/jqplot.js

```diff
@@ -377,6 +377,23 @@ function rowEvolutionGetMetricNameFromRow(tr)
 
             // create jqplot chart
             try {
+
+                // Work out incomplete data points
+                this.jqplotParams['incompleteDataPoints'] = 0;
+                if (this.param.date && this.param.period) {
+
+                    var endDate = null;
+                    if (this.param.date.indexOf(',')) {
+                        var s = this.param.date.split(',')
+                        endDate = s[s.length-1];
+                    } else {
+                        endDate = this.param.date;
+                    }
```

Comment:
This won't work for periods that end today or in the future. The `date` param is often not the end of the period range. I guess the easiest solution would be to use the [piwikPeriods service](https://github.com/matomo-org/matomo/blob/2c30dc8d231af58cdf2516a413da2d86bf7d62c6/plugins/CoreHome/angularjs/common/services/periods.js) to parse the period/date and then use the end date for comparing

Reply:
Thanks for the tip, I didn't know about piwikPeriods, much easier!